src/tasks/get-items-by-category-task.ts

```diff
@@ -27,12 +28,17 @@ export class getItemsByCategoriesTask extends BaseCategoryTask<Item[]> {
     this.status = 'RUNNING';
 
     const { categoryIds } = this.input;
-    this.targetId = categoryIds?.join(',');
-    const items = await this.categoryService.getItemsByCategories(
-      categoryIds,
-      handler,
-    );
+    
+    const itemIdsList = await Promise.all(categoryIds.map(async (idString) => {
+      const categoryIdList = idString.split(',');
+      const itemIds = await this.categoryService.getItemsByCategories(
+        categoryIdList,
+        handler
+      );
+      return itemIds;
+    }));
 
+    const items = itemIdsList.length > 1 ? itemIdsList.reduce((a, b) => a.filter((c) => getItemIdsArray(b).includes(c.id))) : itemIdsList[0];
```

Comment:
I might be wrong, but the first clause might work even tho there's one entry? To test.

Reply:
Add comments.

Reply:
And could you test your implementation? Add a test with at least two categories, test if the intersection works, mainly. This should go in `get-items-by-cateory-task.test.ts` in the same directory.